feat: add zinter command

ClaudiuCeia opened the pull request:
Hi there,

This PR adds the [zinter](https://redis.io/commands/zinter/) command which was missing. I tried to follow the same patterns I saw used for the other commands. Fun fact I learned today - `sinter` also works with sorted sets, but there's not `WITHSCORES` options which is what I needed. 

Cheers

uki00a commented on command.ts:
```diff
@@ -933,6 +937,8 @@ XRANGE somestream - +
     member: RedisValue,
   ): Promise<BulkString>;
+  zinter(keys: string[], opts?: ZInterOpts): Promise<Raw[]>;
+  zinter(key_weights: [string, number][], opts?: ZInterOpts): Promise<Raw[]>;
```
Could you add the following overload signature to support `Record<string, number>`?

```suggestion
  zinter(key_weights: [string, number][], opts?: ZInterOpts): Promise<Raw[]>;
  zinter(key_weights: Record<string, number>, opts?: ZInterOpts): Promise<Raw[]>;
```

uki00a reviewed (changes requested):
@ClaudiuCeia Thanks for the PR!

Could you run the following command to update `mod.ts`?

```shell
$ deno task make_mod
```

This should resolve CI failure

ClaudiuCeia commented:
Hi @uki00a 

Thanks for the review! I ran `make_mod` and added the signature you requested. Looks like my editor also removed some trailing commas, I'll revert those now. 

Is there any way to ensure consistency here - do you use any kind of setup here that I should be aware of?

uki00a commented:
@ClaudiuCeia Thanks for the making changes! We run `deno fmt --check` in CI to keep the code style consistent. Could you run `deno fmt` locally on the changed files?

uki00a commented:
@ClaudiuCeia Thank you for the contribution! I'll release this later
